src/scripts/main.js

```diff
@@ -1,11 +1,28 @@
 'use strict';
 
+const root = document.querySelector('body');
+
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(
+      eventName,
+      () => {
+        resolve(
+          `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+        );
+      },
+      { once: true },
+    );
+  });
 }
 
 const printMessage = (message) => {
-  // write your code here
+  const messageNode = document.createElement('div');
+
+  messageNode.classList.add('message');
+  messageNode.textContent = message;
+
+  root.append(message);
```

Comment:
You're trying to append a string to the DOM, but you should be appending a node. Change `root.append(message)` to `root.append(messageNode)`.